src/Components/Components/src/Reflection/PropertySetter.cs

```diff
@@ -17,6 +17,10 @@ internal sealed class PropertySetter
         "ReflectionAnalysis",
         "IL2060:MakeGenericMethod",
         Justification = "The referenced methods don't have any DynamicallyAccessedMembers annotations. See https://github.com/mono/linker/issues/1727")]
+    [SuppressMessage(
+        "AOT",
+        "IL3050:Calling members annotated with 'RequiresDynamicCodeAttribute' may break functionality when AOT compiling.",
+        Justification = "The referenced methods are AOT safe with reference types.")]
```

Comment:
Is this only used with reference types?

Add a `Debug.Assert(targetType.IsClass)` and `Debug.Assert(property.PropertyType.IsClass)` to verify.

Reply:
You've used SuppressMessage instead of UnconditionalSuppressMessage throughout this PR. Is there a reason why?

Reply:
> You've used SuppressMessage instead of UnconditionalSuppressMessage throughout this PR. Is there a reason why?

Nope, just what VS suggested. I'll fix those and add the assertions.